docs/content/docs/dev/table/hiveCompatibility/hiveserver2.md

```diff
@@ -0,0 +1,306 @@
+        </tr>
+        <tr>
+            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.host</h5></td>
+            <td>optional</td>
+            <td style="word-wrap: break-word;">(none)</td>
+            <td>String</td>
+            <td>The server address of HiveServer2 host to be used for communication.Default is empty, which means the to bind to the localhost. This is only necessary if the host has multiple network addresses.</td>
+        </tr>
+        <tr>
+            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.login.timeout</h5></td>
+            <td>optional</td>
+            <td style="word-wrap: break-word;">20 s</td>
+            <td>Duration</td>
+            <td>Timeout for Thrift clients during login to HiveServer2</td>
+        </tr>
+        <tr>
+            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.max.message.size</h5></td>
+            <td>optional</td>
+            <td style="word-wrap: break-word;">104857600</td>
+            <td>Long</td>
+            <td>Maximum message size in bytes a HS2 server will accept.</td>
+        </tr>
+        <tr>
+            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.port</h5></td>
+            <td>optional</td>
+            <td style="word-wrap: break-word;">10000</td>
+            <td>Integer</td>
+            <td>The port of the HiveServer2 endpoint.</td>
+        </tr>
+        <tr>
+            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.worker.keepalive-time</h5></td>
+            <td>optional</td>
+            <td style="word-wrap: break-word;">1 min</td>
+            <td>Duration</td>
+            <td>Keepalive time for an idle worker thread. When the number of workers exceeds min workers, excessive threads are killed after this time interval.</td>
+        </tr>
+        <tr>
+            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.worker.threads.max</h5></td>
+            <td>optional</td>
+            <td style="word-wrap: break-word;">512</td>
+            <td>Integer</td>
+            <td>The maximum number of Thrift worker threads</td>
+        </tr>
+        <tr>
+            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.worker.threads.min</h5></td>
+            <td>optional</td>
+            <td style="word-wrap: break-word;">5</td>
+            <td>Integer</td>
+            <td>The minimum number of Thrift worker threads</td>
+        </tr>
+    </tbody>
+</table>
+
+Features
+----------------
+
+### Built-in HiveCatalog and Hive Dialect
+
+The SQL Gateway with HiveServer2 endpoint aims to provide the same experience compared to the HiveServer2. When users connect to the HiveServer2,
+the HiveServer2 endpoint creates the Hive Catalog as the default catalog, switches to the Hive dialect, and executes the SQL in batch mode for the session.
+Users can submit the Hive SQL in Hive style but execute it in the Flink environment.
+
+### Integrate into the Hive Ecosystem
```

Comment:
Rename the section name to "Clients" and use H2 titles to display the supported clients in TOC.